chingu-voyages/v52-tier3-team-38
# Created bookAppointment API for residents to schedule appointments

# Description
This PR adds an API to handle appointment booking for residents.

## Related to the following Jira/Trello issue 
Create a database to store resident contact information.

- Insert link to a relevant task/issue in Jira/Trello
- https://trello.com/c/cM8UbjdC/40-create-a-database-to-store-resident-contact-information

## Main changes explained:
- Created bookAppointment API for the reason listed above.
- Adjusted Appointment schema to fix bug that checked for the existence of a resident model instead of appointment.

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` to run this PR locally
3. To test an API I would recommend installing POSTMAN or something similar. 
4. Create a POST request to http://localhost:3000/api/user/ *insert id of a user in the database here*/bookAppointment. The body of the request should include the timeslot and the address such as this {
    "timeslot": "2024-11-23 15:00:00",
    "address": "3445 Sydnee Meadow"
}
7. A successful request should return a success message and the newly created appointment.
8. Test error handling to verify they work as intended.


## Note:
The first 3 users in the database, Elisa, Maurice, and Sherman all have existing appointments. You test their ids to make sure they cannot have more than one booked appointment. Use any of the others to book an appointment. As always if you have any suggestions to improve the API either now or in the future, please let me know.
